### DISCUSSION TITLE
Building code shared between interception scripts

### DISCUSSION THREAD

**Discussion by tawaren** [JanssenProject/jans#7790] [Q&A] [answered]
Some of the Java interception scripts I have contain duplicated code. To eliminate the duplication, I consider putting a jar file containing the shared code into the folder "/opt/jans/jetty/jans-auth/custom/libs". However, the shared code imports some Janssen classes, and I wonder what the best approach is to build a jar from the shared code. 

The only approach I currently see is to clone the Janssen repo, build with Maven and publish it into the local Maven repository, so that the other build process can load the dependency from there. 

However, I do not like this approach and was wondering if there are better alternatives to reference Janssen as a dependency during building, preferably without the need to build it locally

**Comment by yuriyz** [comment, marked as answer]
There is no need to build it locally. Simply create new maven project and reference jans maven repo https://maven.jans.io/maven .
Just use dependencies from jans repo as you normally do in any maven project.

**Reply by tawaren** [reply]
Thanks, did not realize that you have your own maven repository for the project.
